ensime_shared/client.py

```diff
@@ -83,38 +86,38 @@ def setup_logger():
 
         super(EnsimeClient, self).__init__()
         self.editor = editor
-        self.launcher = launcher
+        self.config = config
 
         self.log = setup_logger()
         self.log.debug('__init__: in')
         self.editor.initialize()
 
         self.ws = None
-        self.ensime = None
-        self.ensime_server = None
+        self.server = None  # TODO: try to get rid of this (reconnect)
 
         self.call_id = 0
         self.call_options = {}
+        self.debug_thread_id = None
         self.refactor_id = 1
         self.refactorings = {}
 
         # Queue for messages received from the ensime server.
         self.queue = Queue()
+
         self.suggestions = None
         self.completion_timeout = 10  # seconds
         self.completion_started = False
 
         self.full_types_enabled = False
         """Whether fully-qualified types are displayed by inspections or not"""
 
-        self.toggle_teardown = True
-        self.connection_attempts = 0
         self.tmp_diff_folder = tempfile.mkdtemp(prefix='ensime-vim-diffs')
 
         # By default, don't connect to server more than once
```

Comment:
No it doesn't, thanks for pointing that out—I still want to make this stuff clearer. `connection_retries` was a rename from `number_try_connection` because it was clearer to me for how it was used, but it's still confusing that the `connect` method and `queue_poll` are based on counting down `connection_retries`, then what is now `watchdog_client_server_connection` counts *up* `connection_attempts`. I'm guessing that at one time the former was meant to be like a config value for max retries, and the count-up should use that instead of a hardcoded literal, but it perhaps got mixed up over time.

I was pretty much translating the existing code on these up until now, but I'll consolidate them.